demo/apisix/tmail-apisix-plugin-runner/src/main/java/com/linagora/apisix/plugin/RedisRevokedTokenRepository.java

```diff
@@ -2,25 +2,51 @@
 
 import java.time.Duration;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.lettuce.core.RedisException;
 import io.lettuce.core.SetArgs;
 import io.lettuce.core.api.sync.RedisStringCommands;
 
 public class RedisRevokedTokenRepository implements IRevokedTokenRepository {
+
     public static final Duration CACHE_DURATION = Duration.ofDays(7);
+    public static final Boolean IGNORE_REDIS_ERRORS = true;
+    private final Logger logger = LoggerFactory.getLogger("RevokedTokenPlugin");
 
     private final RedisStringCommands<String, String> redisCommand;
 
-    public RedisRevokedTokenRepository(RedisStringCommands<String, String> redisCommand) {
+    private final boolean ignoreRedisErrors;
+
+    public RedisRevokedTokenRepository(RedisStringCommands<String, String> redisCommand,
+                                       boolean ignoreRedisErrors) {
         this.redisCommand = redisCommand;
+        this.ignoreRedisErrors = ignoreRedisErrors;
     }
 
     @Override
     public void add(String sid) {
-        redisCommand.set(sid, Boolean.TRUE.toString(), SetArgs.Builder.ex(CACHE_DURATION));
+        try {
+            redisCommand.set(sid, Boolean.TRUE.toString(), SetArgs.Builder.ex(CACHE_DURATION));
+        } catch (RedisException e) {
+            if (!ignoreRedisErrors) {
+                throw e;
+            }
+            logger.warn("Error while add revoked token in Redis: {}", e.getMessage());
```

Comment:
Don't log-and-throw

Either log or throw

Reply:
That's the way it is right now, it will "Either log or throw"

Reply:
Sorry; not woken up well...